File: music-lessons-2-java/app/src/main/java/com/mstoyanov/musiclessons/repository/PhoneNumberDao.java
package com.mstoyanov.musiclessons.repository;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import com.mstoyanov.musiclessons.model.PhoneNumber;

import java.util.List;

@Dao
public interface PhoneNumberDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAll(List<PhoneNumber> phoneNumbers);

    @Query("select * from phone_number where student_id = :studentId")
    List<PhoneNumber> findAllByStudentId(long studentId);

    @Delete
    void delete(PhoneNumber phoneNumber);
}

File: music-lessons-2-kotlin/app/src/main/java/com/mstoyanov/musiclessons/repository/PhoneNumberDao.kt
package com.mstoyanov.musiclessons.repository

import android.arch.persistence.room.*
import com.mstoyanov.musiclessons.model.PhoneNumber

@Dao
interface PhoneNumberDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAll(phoneNumbers: List<PhoneNumber>)

    @Query("select * from phone_number where student_id = :arg0")
    fun findAllByStudentId(studentId: Long): MutableList<PhoneNumber>

    @Delete
    fun delete(phoneNumber: PhoneNumber)
}